app/src/main/java/ch/hikemate/app/model/route/HikesViewModel.kt

```diff
@@ -391,9 +451,39 @@ class HikesViewModel(
    * [_hikeFlowsMap] is the one that actually gets updated during operations. Once it has been
    * updated, we need to update [_hikeFlowsList] to reflect the changes. This is what this helper
    * function does.
+   *
+   * Because [_hikeFlowsList] is updated, this function also updates [_allOsmDataLoaded] to reflect
+   * whether all hikes in [_hikeFlowsList] have their OSM data loaded.
    */
-  private fun updateHikeFlowsList() {
+  private fun updateHikeFlowsListAndOsmDataStatus() {
     _hikeFlowsList.value = _hikeFlowsMap.values.toList()
+    updateOsmDataAvailabilityStatus()
+  }
+
+  /**
+   * Internal helper function.
+   *
+   * The exposed [allOsmDataLoaded] indicates whether all loaded hikes in [hikeFlows] have their OSM
+   * data available. This needs to be updated when either the entire list is updated, or a single
+   * hike is updated.
+   *
+   * This helper function updates the value of [_allOsmDataLoaded] accordingly to [_hikeFlowsList].
+   */
+  private fun updateOsmDataAvailabilityStatus() {
+    _allOsmDataLoaded.value = _hikeFlowsList.value.all { it.value.hasOsmData() }
+  }
+
+  /**
+   * Internal helper function.
+   *
+   * Sets [_loadedHikesType] with the new provided value if not already equal to that value.
+   *
+   * Requires [_hikesMutex] to have been acquired by the caller.
+   */
+  private fun setLoadedHikesType(value: LoadedHikes) {
+    if (_loadedHikesType.value != value) {
+      _loadedHikesType.value = value
+    }
```

Comment:
Question: I am curious as to why you check if `_loadedHikesType` isn't equal to the `value` given as parameter?

Reply:
I'm not sure about this, but from what I've seen, setting a state flow to the same value as its previous will still launch UI updates, even though unnecessary.

So to avoid unnecessary updates I check the current value first.

Reply:
Yes, you are right I looked it up and even if it's the same value it will trigger a re composition.